### PR TITLE
MUL-804: bring back transaction_update function

### PR DESCRIPTION
**Pull request by Enmk**
Added API call transaction_update()
Added tests for transaction_update().

### CODE REVIEW

File: multy_test/test_transaction.cpp [hunk 1]
```diff
@@ -217,6 +222,14 @@ GTEST_TEST(TransactionTestInvalidArgs, transaction_serialize)
     EXPECT_NE(nullptr, error);
 }
 
+GTEST_TEST(TransactionTestInvalidArgs, transaction_update)
+{
+    TestTransaction transaction;
+    BinaryDataPtr binary_data;
```
Comment (PashaKlybik) [inline multy_test/test_transaction.cpp]: Why is this here?

File: multy_test/test_bitcoin_transaction.cpp [hunk 2]
```diff
@@ -590,3 +685,62 @@ GTEST_TEST(BitcoinTransactionTest, SmokeTest_with_many_input_from_different_addr
...
+    HANDLE_ERROR(transaction_update(transaction.get()));
+
+    BigIntPtr updated_total_fee;
+    HANDLE_ERROR(transaction_get_total_fee(transaction.get(), reset_sp(updated_total_fee)));
+    EXPECT_NE("0", *updated_total_fee);
+
+    BinaryDataPtr serialized;
+    HANDLE_ERROR(transaction_serialize(transaction.get(), reset_sp(serialized)));
+
+    EXPECT_LT(*updated_total_fee, *total_fee);
```
Comment (PashaKlybik) [inline multy_test/test_bitcoin_transaction.cpp]: Add comment what this do/

**Review by PashaKlybik — CHANGES_REQUESTED** [review]

File: multy_test/test_transaction.cpp (hunk 1 above)
Comment (Enmk) [reply multy_test/test_transaction.cpp]: Removed

File: multy_test/test_bitcoin_transaction.cpp (hunk 2 above)
Comment (Enmk) [reply multy_test/test_bitcoin_transaction.cpp]: Fixed

**Review by PashaKlybik — APPROVED** [review]
All good!
